service/history/hsm/events.go

```diff
@@ -44,5 +44,5 @@ type EventDefinition interface {
 	// Command events should never be cherry picked as we rely on the workflow to reschedule them.
 	// Return [ErrNotCherryPickable], [ErrStateMachineNotFound], or [ErrInvalidTransition] to skip cherry picking. Any
 	// other error is considered fatal and will abort the cherry pick process.
-	CherryPick(root *Node, event *historypb.HistoryEvent) error
+	CherryPick(root *Node, event *historypb.HistoryEvent, resetReapplyExcludeTypes map[enumspb.ResetReapplyExcludeType]bool) error
```

Comment:
I would really prefer if this was a `map[enumspb.ResetReapplyExcludeType]struct{}`, which is the go way of defining sets.

Reply:
Agree. I [mentioned this in another thread](https://github.com/temporalio/temporal/pull/6598#discussion_r1792851807) but resolved it before you saw it I think :).
The caller already holds the set in this type [here](https://github.com/temporalio/temporal/blob/main/service/history/ndc/workflow_resetter.go#L740). So either
- I change that type which will cascade into 4 level of callers higher and into few other places.
- convert from `map[x]bool` to `map[x]struct{}` just before this call.
I think converting just for this function call is not worth it. so decided to stick with map[]bool. Happy to convert to map[]struct end-to-end in another PR.

Reply:
Would rather convert everywhere, not just here. Can wait for a follow up PR though.